webui/react/src/components/HyperparameterSearchModal.tsx

```diff
@@ -117,6 +117,12 @@ const HyperparameterSearchModal = ({ closeModal, experiment, trial }: Props): JS
     });
   }, [experiment.hyperparameters, trialHyperparameters]);
 
+  const [currentHPs, setCurrentHPs] =
+    useState<{ hyperparameter: Hyperparameter; name: string }[]>();
+  useEffect(() => {
+    !currentHPs && hyperparameters && setCurrentHPs(hyperparameters);
+  }, [hyperparameters, currentHPs]);
+
```

Comment:
sugg: this useEffect is only being used to set the initial state, and since `hyperparameters` isn't referred to elsewhere, I believe this whole section can be rewritten to something like:
```
const calculateInitialHyperparameters = useCallback(() => {
    return Object.entries(experiment.hyperparameters).map((hp) => {
      const hpObject = { hyperparameter: hp[1], name: hp[0] };
      if (trialHyperparameters?.[hp[0]]) {
        hpObject.hyperparameter.val = trialHyperparameters[hp[0]];
      }
      return hpObject;
    });
  }, [experiment.hyperparameters, trialHyperparameters]);

const [currentHPs, setCurrentHPs] =
    useState<{ hyperparameter: Hyperparameter; name: string }[]>(calculateInitialHyperparameters);
```